contracts/eternal-storage/EternalStorage.sol

```diff
@@ -14,21 +14,25 @@ contract EternalStorage {
     // Address of the current implementation
     address internal _implementation;
 
+    // Address of the owner of the contract
+    address internal _owner;
+
+    /**
+     * @dev Access check: ensure that either we are on the genesis block, or
+     * `msg.sender` is the owner of the contract.  The genesis block is
+     * hard-coded into the client, so attacks based on it are not possible.
+     */
+    modifier onlyOwner() {
```

Comment:
It's better to move this modifier to some inherited file called `Owned`, for example. I think we shouldn't mix the file containing the eternal state mappings with modifiers or functions.

Reply:
@varasev should `OwnedEternalStorage` inherit from `EternalStorage`?  It seems to me that the reverse would be clearer.

Reply:
We can leave the `_owner` in the `EternalStorage`, but move the `onlyOwner`  to `eternal-storage/OwnedEternalStorage.sol` which is inherited from `EternalStorage`. And then change inheritance for the upgradable contracts so that they are inherited from `OwnedEternalStorage` (not from `EternalStorage`). These contracts are `BlockRewardBase`, `RandomBase`, `ValidatorSetBase`, `Certifier`, `KeyGenHistory`, `TxPermission`, and `EternalStorageProxy`.

Reply:
I created a separate base class `Owned` instead.  Is that acceptable?

Reply:
> I created a separate base class `Owned` instead. Is that acceptable?

No, I explained below why.